shared/src/models/utils/cater-for.ts

```diff
@@ -1,17 +1,35 @@
-import { IsNumber, IsString } from 'class-validator';
+import { IsNumber, IsString, Min } from 'class-validator';
 import { getCreate } from './get-create';
 
 export class CanCaterFor {
+    /**
+     * if {@link logicalOperator}  `=== 'and'` it is cumulative,
+     * how many red patients can the catering treat
+     * also affects the number of possible {@link yellow} and {@link green} patients
+     * that can be treated.
+     */
     @IsNumber()
+    @Min(0)
     public readonly red: number;
 
+    /**
+     * if {@link logicalOperator}  `=== 'and'` it is cumulative,
+     * how many yellow patients can the catering treat
+     * to the number already written in the {@link red} value
+     */
     @IsNumber()
+    @Min(0)
     public readonly yellow: number;
 
+    /**
+     * if {@link logicalOperator}  `=== 'and'` it is cumulative,
+     * how many green patients can the catering treat
+     * to the number already written in the {@link yellow} and {@link red} value
+     */
     @IsNumber()
+    @Min(0)
     public readonly green: number;
 
-    // TODO
     @IsString()
     public readonly logicalOperator: 'and' | 'or';
```

Comment:
@christianzoellner what do you think about removing `or` what would be possible implications of `or`?
I can think of e.g. if a personnel should only be able to treat `red` and `yellow` patients and won't be able (does not want to) treat green patients.

Reply:
> I can think of e.g. if a personnel should only be able to treat `red` and `yellow` patients and won't be able (does not want to) treat green patients.

Not treating / wanting to treat lower-category patient does _not_ make sense! Some people might not like their job, but emergency service personell is meant to be deployed flexibly, and someone must also care for the green patients (who often nevertheless have hidden injuries and might become critical later on, so one should not withhold medical competence from them) [1]. 

Whether the `or` operator can be sufficiently replaced with the `cumulative and` I do not know, since I do not have a definition of the latter. :-)

[1] Actually, there currently is a lot of debate about ambulance services having to deal with minor stuff  for a large percentage of their time ("helping instead of rescuing").

Reply:
@Dassderdie 
@christianzoellner and I talked and he proposed the following:
- every personnel and material has a capacity (he proposed it to be `1.0`, as a float) - this capacity does not differ between patient categories - every personnel can have a different capacity, e.g. `notarzt=1.2` and `san=0.4` or so
- every patient (or in the first step every patient category) has a weight/capacity it consumes from a personnel
- therefore a personnel could e.g. only treat one `red`, but could treat instead three `greens`, as e.g. red takes up `1.0` and `green` only `0.3`

If the old model would be used, `or` should not be removed, as otherwise it is not possible to make that a personnel can only treat 1 `red` but could also treat instead 5 `green` or so.

This model could **later** be changed as that every patient has different weights/capacities he consumes, maybe on the phase he is in.

@christianzoellner If you want to add or correct

Reply:
Thanks for the summary of our discussion @anonym-HPI . 

Let me get to a meta perspective for a moment:
1. Currently we need the `or` operator. It is an important aspect that some patients need more capacity than others (e.g. that one personnel can treat only 1 red patient and do nothing else or alternatively supervise 5 green patients where most of the work is looking out if they get worse and which thus can be done in parallel).
2. On the long term, I propose the "capacity" model that Marvins describes, since it gives us way more flexibility in the future (e.g. by combining treatment capacities with state-dependand patient treatment requirements).

Reply:
Ok. The `and` & `or` logic stays. 
Just be careful not to fall into the trap of trying to make too complex treatment models. (YAGNI, KISS, creeping featurism, ...)

I have added the capacity model to the list [here](https://github.com/hpi-sam/digital-fuesim-manv/issues/489#issuecomment-1245158208).